planner/core/point_get_plan.go

```diff
@@ -1006,3 +1042,73 @@ func (p *PointGetPlan) findHandleCol() *expression.Column {
 	}
 	return handleCol
 }
+
+func buildSchemaForPartition(dbName model.CIStr, tbl *model.TableInfo, tblName model.CIStr) (*expression.Schema, []*types.FieldName) {
+	columns := make([]*expression.Column, 0, len(tbl.Columns)+1)
+	names := make([]*types.FieldName, 0, len(tbl.Columns)+1)
+	for _, col := range tbl.Columns {
+		names = append(names, &types.FieldName{
+			DBName:      dbName,
+			OrigTblName: tbl.Name,
+			TblName:     tblName,
+			ColName:     col.Name,
+		})
+		columns = append(columns, colInfoToColumn(col, len(columns)))
+	}
+	return expression.NewSchema(columns...), names
+}
+
+func locateHashPartition(piExpr expression.Expression, pairs []nameValuePair) (int64, bool) {
+	switch pi := piExpr.(type) {
+	case *expression.Column:
+		for _, p := range pairs {
+			if p.colName == pi.OrigName {
```

Comment:
Can `pi.OrigName` be upper case?

Reply:
I add `OrigName: col.Name.L` in the `colInfoToColumn` function.